src/core/frames/frame_controller.ts

```diff
@@ -277,6 +284,16 @@ export class FrameController implements AppearanceObserverDelegate, FetchRequest
     }
   }
 
+  private visitCachedSnapshot = ({ element }: Snapshot) => {
+    const frame = element.querySelector("#" + this.element.id)
+
+    if (frame && this.previousContents) {
+      frame.innerHTML = ""
+      frame.append(this.previousContents)
```

Comment:
Ideally, this implementation would call [ParentNode.replaceChildren][], but unfortunately, `typescript@4.1.x` doesn't support that API. It's [introduced][] in `typescript@4.4`, which might be worth upgrading to in a separate PR.

[ParentNode.replaceChildren]: https://developer.mozilla.org/en-US/docs/Web/API/Element/replaceChildren
[introduced]: https://github.com/microsoft/TypeScript-DOM-lib-generator/issues/1029#issuecomment-869224737